x-pack/plugins/apm/public/components/routing/templates/apm_service_template/index.tsx

```diff
@@ -97,6 +108,12 @@ function TemplateWithContext({
     [query, router, selectedTab, serviceName, title]
   );
 
+  if (isMobileAgentName(agentName)) {
+    replace(history, {
+      pathname: location.pathname.replace('services', 'mobile-services'),
```

Comment:
Can you make this check more strict?

```suggestion
      pathname: location.pathname.replace('/apm/services/', '/apm/mobile-services/'),
```

Reply:
The tests were failing and I realized that `location.pathname` doesn't' include. I'll make it a bit moe strict like that

```
pathname: location.pathname.replace('/services/', '/mobile-services/'),
```